File: FrontEnd/frontAdset/src/app/interfaces/Veiculo/VeiculoDto.ts
import { FotoDto } from "../Foto/FotoDto";
import { PacotePortalDto } from "../Pacote/PacotePortalDto";
import { OpcionalVeiculoDto } from "../Veiculo/OpcionalVeiculoDto";
export interface VeiculoDto {
  id: number;
  marca: string;
  modelo: string;
  ano: number;
  placa: string;
  km: number | null;
  cor: string;
  preco: number;
  fotos: FotoDto[]; 
  opcionais: OpcionalVeiculoDto[];
  pacotesPortal: PacotePortalDto[];
}

File: BackEnd/API/Dtos/Veiculo/VeiculoDto.cs


using API.Dtos.Foto;
using API.Dtos.RelacaoVeiculoOpcional;
using API.Dtos.RelacaoVeiculoPacotePortal;

namespace API.Dtos.Veiculo
{
    public class VeiculoDto
    {

        public int Id { get; set; }
        public string Marca { get; set; } = null!;
        public string Modelo { get; set; } = null!;
        public int Ano { get; set; }
        public string Placa { get; set; } = null!;
        public int? Km { get; set; }
        public string Cor { get; set; } = null!;
        public decimal Preco { get; set; } 

        public List<FotoDto> Fotos { get; set; } = new();
        public List<RelacaoVeiculoOpcionalDto> Opcionais { get; set; } = new();
        public List<RelacaoVeiculoPacotePortalDto> PacotesPortal { get; set; } = new();

    }


}
